misc: Use rommapp/romm-devcontainer image in docker-compose

zurdi15 opened the pull request:
<!-- trunk-ignore-all(markdownlint/MD041) -->
<!-- trunk-ignore-all(markdownlint/MD033) -->

**Description**
Added a new [repo](https://hub.docker.com/repository/docker/rommapp/romm-devcontainer/general) in Dockerhub to upload a devcontainer image to avoid contributors to build it locally.

**Checklist**
<sup>Please check all that apply.</sup>

- [x] I've tested the changes locally
- [x] I've updated relevant comments
- [x] I've assigned reviewers for this PR
- [ ] I've added unit tests that cover the changes

gantoine reviewed (commented):
This means that any changes to package.json or uv packages will require building and pushing a new image no? And should we be versioning this image to match the latest release?

zurdi15 commented:
> This means that any changes to package.json or uv packages will require building and pushing a new image no? And should we be versioning this image to match the latest release?

hmm yes, any dependency change will need a new build + push. About versioning it the same way than the main image that's a great idea to know which can be used to develop which version

zurdi15 committed:
Add Dockerfile and entrypoint script for development environment setup

- Create .devcontainer/Dockerfile for frontend and Python dependencies
- Create .devcontainer/Dockerfile.base for system dependencies
- Add entrypoint script to manage service startup and symlink creation
- Update docker-compose.yml to reference the new Dockerfile
- Modify DEVELOPER_SETUP.md for updated installation instructions

zurdi15 committed:
Add trunk-ignore comments for hadolint in Dockerfile.base

gantoine reviewed (changes requested):
🔥🔥🔥

zurdi15 committed:
Remove unnecessary empty lines from DEVELOPER_SETUP.md
